index.js

```diff
@@ -1270,18 +1304,32 @@ function checkCode(context) {
     const readFiles = [
         '.npmignore',
         '.gitignore',
-        'admin/index_m.html',
         'iob_npm.done',
-        'admin/words.js',
-        '.travis.yml',
-        'jsonCustom.json'
+        '.travis.yml'
     ];
     if (context.packageJson.main) {
         readFiles.push(context.packageJson.main);
     }
 
+    if (!context.ioPackageJson.common.noConfig) {
+        if (context.ioPackageJson.common.materialize || (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.config === 'materialize')) {
+            readFiles.push('admin/index_m.html');
+            readFiles.push('admin/words.js');
+        }
+
+        if (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.config === 'json') {
+            readFiles.push('admin/jsonConfig.json');
+            readFiles.push('admin/i18n/en/translations.json');
+        }
+
+        if (context.ioPackageJson.common.jsonCustom || (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.custom === 'json')) {
```

Comment:
Maybe only check for "supportsCustom" because admin5 requires (hard) to have jsonConfig for customs